Lib/compression/zstd/__init__.py

```diff
@@ -0,0 +1,293 @@
+    if not isinstance(dict_size, int):
+        raise TypeError('dict_size argument should be an int object.')
+    if not isinstance(level, int):
+        raise TypeError('level argument should be an int object.')
+
+    # Prepare data
+    chunks = []
+    chunk_sizes = []
+    for chunk in samples:
+        chunks.append(chunk)
+        chunk_sizes.append(_nbytes(chunk))
+
+    chunks = b''.join(chunks)
+    if not chunks:
+        raise ValueError("The samples are empty content, can't finalize dictionary.")
+
+    # custom_dict_bytes: existing dictionary.
+    # samples_bytes: samples be stored concatenated in a single flat buffer.
+    # samples_size_list: a list of each sample's size.
+    # dict_size: maximal size of the dictionary, in bytes.
+    # compression_level: compression level expected to use in production.
+    dict_content = _finalize_dict(zstd_dict.dict_content,
+                                  chunks, chunk_sizes,
+                                  dict_size, level)
+
+    return _zstd.ZstdDict(dict_content)
+
+def compress(data, level=None, options=None, zstd_dict=None):
+    """Compress a block of data, return a bytes object of zstd compressed data.
+
+    Refer to ZstdCompressor's docstring for a description of the
+    optional arguments *level*, *options*, and *zstd_dict*.
+
+    For incremental compression, use an ZstdCompressor instead.
+    """
+    comp = ZstdCompressor(level=level, options=options, zstd_dict=zstd_dict)
+    return comp.compress(data, ZstdCompressor.FLUSH_FRAME)
+
+def decompress(data, zstd_dict=None, options=None):
+    """Decompress one or more frames of data.
+
+    Refer to ZstdDecompressor's docstring for a description of the
+    optional arguments *zstd_dict*, *options*.
+
+    For incremental decompression, use an ZstdDecompressor instead.
+    """
+    results = []
+    while True:
+        decomp = ZstdDecompressor(options=options, zstd_dict=zstd_dict)
+        try:
+            res = decomp.decompress(data)
+        except ZstdError:
+            if results:
+                break  # Leftover data is not a valid LZMA/XZ stream; ignore it.
+            else:
+                raise  # Error on the first iteration; bail out.
+        results.append(res)
+        if not decomp.eof:
+            raise ZstdError("Compressed data ended before the "
+                            "end-of-stream marker was reached")
+        data = decomp.unused_data
+        if not data:
+            break
+    return b"".join(results)
+
+class _UnsupportedCParameter:
+    def __set_name__(self, _, name):
+        self.name = name
+
+    def __get__(self, *_, **__):
+        msg = ("%s CParameter not available, zstd version is %s.") % (
+            self.name,
+            zstd_version,
+        )
+        raise NotImplementedError(msg)
+
+
+class CParameter(enum.IntEnum):
+    """Compression parameters"""
+
+    compressionLevel = _zstd._ZSTD_c_compressionLevel
```

Comment:
How required is the camelCase naming? As per above, for new code we should avoid non-standard names.